public/apps/configuration/utils/internal-user-list-utils.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import { map } from 'lodash';
+import { HttpStart } from '../../../../../../src/core/public';
+import { API_ENDPOINT_INTERNALUSERS } from '../constants';
+import { DataObject, InternalUser, ObjectsMessage } from '../types';
+
+export interface InternalUsersListing extends InternalUser {
+  username: string;
+}
+
+export function transformUserData(rawData: DataObject<InternalUser>): InternalUsersListing[] {
+  return map(rawData, (value: InternalUser, key?: string) => ({
+    username: key || '',
+    attributes: value.attributes,
+  }));
+}
+
+export async function requestDeleteUsers(http: HttpStart, users: string[]) {
+  for (const user of users) {
+    await http.delete(`${API_ENDPOINT_INTERNALUSERS}/${user}`);
```

Comment:
Just curious if one user fails to delete, will it still try to run deletion for the remaining users or stop wherever the first failure happens.

Reply:
It stops upon first failure.

Reply:
Seems a better experience to do best effort to delete the remaining. I will leave it to you to decide.

Reply:
In my opinion on the frontend should we keep it simple. It would be better to have a batch deletion API in the future. So I didn't do any extra handling here.